### ISSUE TITLE
Basket update: which fields needs to be changed or added?

### ISSUE THREAD

**Issue by patjouk** [mozilla/donate-wagtail#115]
After discussing with Samir on Slack about #27, it was decided to first update basket and then connect `donate-wagtail` to it. To do that, we need to know what fields needs to be updated/added/remove.

Curent JSON payload sent to basket:
`first_name`: first name of the donor (optional),
`last_name`: last name of the donor,
`email`: email of the donor,
`donation_amount`: amount of the donation,
`currency`: letter code for the currency,
`created`: unix timestamp,
`recurring`: `false` for a one-time donation or `true` for a recurring one,
`service`: name of the payment processor (ex: `stripe` or `paypal`)
`transaction_id`: ID generated by the payment processor,
`project`: name of the project that will receive the donation (ex: `thunderbird`, `mozillafoundation`)
`last_4`: last 4 digits of the credit card,
`donation_url`: url from which the donation was made,
`locale`: language code for the donor,
`conversion_amount`: donation amount in USD, before transaction fees,
`net_amount`: donation amount in USD, after transaction fees,
`transaction_fee`: payment processor's transaction fees in USD.

@solarissmoke Can you review those fields? Meanwhile I'll go and ask fundraising about the transaction fees.

**Comment by solarissmoke** [comment]
Looking at what [information Braintree provides for a transaction](https://developers.braintreepayments.com/reference/response/transaction/python), I don't think we will be able to determine the transaction fee (we get the Paypal transaction fee, but not Braintree's own fee for card payments) and therefore the net amount. This means `net_amount` and `transaction_fee` are problematic.

The rest of the fields look fine.

**Comment by alanmoo** [comment]
Ah, I got confused about where this comment was @solarissmoke. I’ve reached out to Braintree about those values.

**Comment by alanmoo** [comment]
We are indeed OK to drop `net_amount` and `transaction_fee`; Both Steph and Will agree that it's not in scope of the CRM to keep track of that data.

**Comment by cadecairos** [comment]
I ran some integration testing with Stephanie and Paul, and found the following issues with the current format of data being sent into the message queue.

- Basket currently expects the payload to be wrapped in a `data` attribute, i.e. `{"data":{...}}`
- Basket expects the payload to specify an `event_type` attribute, that it uses to invoke the appropriate worker method. For donations, it is `donation`, and for newsletter signups it is `newsletter_signup_data`. It's a bit strange in retrospect, but when the data object doesn't have an `event_type`, it has a default method it runs through to process things like refunds and disputes (very Stripe specific things, which may not apply here)
- The donation amount is being passed as `amount`, but basket is looking for `donation_amount`
- Basket is looking for a `service` field, which previously was used to differentiate Stripe and Paypal donations
- Basket is expecting the payment frequency to be indicated via a `recurring` boolean attribute. This might need to be changed if we're looking to open up the frequency options (single, monthly, yearly?)
- We're passing the donor's address, unless we absolutely need it in the CRM, we should leave that data in the payment processor and not transmit it.


Here's an example of a payload we are sending now:

```json
{
    "braintree_nonce": "tokencc_asdf_asdf_asdf",
    "amount": "50",
    "landing_url": "https://donate-wagtail-staging.herokuapp.com/en-CA/",
    "project": "mozillafoundation",
    "campaign_id": "",
    "first_name": "Chris",
    "last_name": "DeCairos",
    "email": "cade@example.com",
    "address_line_1": "125 Fake Lane",
    "town": "Toronto",
    "post_code": "F4F4F4",
    "country": "CA",
    "transaction_id": "qy24t6em",
    "last_4": "1111",
    "payment_method": "card",
    "currency": "usd",
    "payment_frequency": "single",
    "payment_method_token": "bbjdw9",
    "source_page_id": 3,
    "locale": "en-CA"
}
```

Here's what it should approximately look like:

```json
{
    "data": {
        "event_type": "donation",
        "braintree_nonce": "tokencc_bd_x78pt7_sx69hz_wsjqgr_9s962d_npz",
        "donation_amount": "50",
        "landing_url": "https://donate-wagtail-staging.herokuapp.com/en-CA/",
        "project": "mozillafoundation",
        "campaign_id": "",
        "first_name": "Chris",
        "last_name": "DeCairos",
        "email": "cade@example.com",
        "transaction_id": "qy24t6em",
        "last_4": "1111",
        "payment_method": "card",
        "currency": "usd",
        "payment_frequency": "single",
        "payment_method_token": "bbjdw9",
        "source_page_id": 3,
        "locale": "en-CA"
    }
}
```

**Comment by alanmoo** [comment]
I believe this can be closed in favor of #455 as the remaining item (and #456 to some extent)
